### PR TITLE
add cobra

### LINKED ISSUE #14: Use a CLI lib (cobra) for the tools
Tools are currently just hacked together. Let's use cobra to provide more structure, and to use flags for the global vars

### PR DESCRIPTION
**Pull request by zramsay**
closes #14

### CODE REVIEW

File: cmd/stake-dist/main.go [hunk 1]
```diff
@@ -2,24 +2,43 @@ package main
...
+
 func main() {
-	args := os.Args[1:]
+	if err := RootCmd.Execute(); err != nil {
+		panic(err)
+	}
+}
+
+func getDist(cmd *cobra.Command, args []string) {
+	node := tmclient.NewHTTP("localhost:26657", "/websocket")
```
Comment (thanethomson) [inline cmd/stake-dist/main.go]: Shouldn't this be:
```suggestion
	node := tmclient.NewHTTP(fullNodeURL, "/websocket")
```

File: cmd/stake-dist/main.go [hunk 2]
```diff
@@ -2,24 +2,43 @@ package main
...
+
+	"github.com/spf13/cobra"
 )
 
 var (
 	cdc = gaia.MakeCodec()
 
-	// expects a locally running node
-	node = tmclient.NewHTTP("localhost:26657", "/websocket")
+	fullNodeUrl string
```
Comment (thanethomson) [inline cmd/stake-dist/main.go]: Rather use the fully capitalized version of the variable name, like so:
```suggestion
	fullNodeURL string
```

File: cmd/stake-dist/main.go [hunk 3]
```diff
@@ -2,24 +2,43 @@ package main
...
 )
 
+func init() {
+	RootCmd.PersistentFlags().StringVarP(&fullNodeUrl, "url", "", "localhost:26657", "URL of synced full-node to use.")
```
Comment (thanethomson) [inline cmd/stake-dist/main.go]:
```suggestion
	RootCmd.PersistentFlags().StringVarP(&fullNodeURL, "url", "", "localhost:26657", "URL of synced full-node to use.")
```

File: cmd/delegation/main.go [hunk 4]
```diff
@@ -13,26 +13,44 @@ import (
...
 
-	// expects a locally running node
-	node = tmclient.NewHTTP("localhost:26657", "/websocket")
-
-	// expects the gos.json file to be here
-	gosJSON = "data/gos.json"
-
-	// output unsigned delegation tx
-	outputFile = "unsigned-delegations.json"
+	fullNodeUrl string
```
Comment (thanethomson) [inline cmd/delegation/main.go]: Rather use the fully capitalized version of the variable name: 
```suggestion
	fullNodeURL string
```

File: cmd/delegation/main.go [hunk 5]
```diff
@@ -13,26 +13,44 @@ import (
...
+	gosJSON     string
+	outputFile  string
 
 	// gas per msg included in the tx
 	gasPerMsg = 200000
 )
 
+func init() {
+	RootCmd.PersistentFlags().StringVarP(&fullNodeUrl, "url", "", "localhost:26657", "URL of synced full-node to use.")
```
Comment (thanethomson) [inline cmd/delegation/main.go]:
```suggestion
	RootCmd.PersistentFlags().StringVarP(&fullNodeURL, "url", "", "localhost:26657", "URL of synced full-node to use.")
```

File: cmd/delegation/main.go [hunk 6]
```diff
@@ -13,26 +13,44 @@ import (
...
 func main() {
-	args := os.Args[1:]
+	if err := RootCmd.Execute(); err != nil {
+		panic(err)
+	}
+}
+
+func getDelegation(cmd *cobra.Command, args []string) {
+
+	node := tmclient.NewHTTP(fullNodeUrl, "/websocket")
```
Comment (thanethomson) [inline cmd/delegation/main.go]:
```suggestion
	node := tmclient.NewHTTP(fullNodeURL, "/websocket")
```

File: cmd/delegation/main.go [hunk 7]
```diff
@@ -130,6 +148,8 @@ func main() {
 }
 
 func getGoSEligibleVals(maxStaked float64, gosMap map[string]float64, validators []stakingtypes.Validator) ([]stakingtypes.Validator, []stakingtypes.Validator) {
+	node := tmclient.NewHTTP(fullNodeUrl, "/websocket")
```
Comment (thanethomson) [inline cmd/delegation/main.go]:
```suggestion
	node := tmclient.NewHTTP(fullNodeURL, "/websocket")
```

**Review by thanethomson — CHANGES_REQUESTED** [review]

File: cmd/stake-dist/main.go (hunk 1 above)
Comment (zramsay) [reply cmd/stake-dist/main.go]: good catch, thanks

**Pull request merged**